istioctl/cmd/metrics.go

```diff
@@ -213,25 +207,43 @@ func metrics(promAPI promv1.API, workload string, duration time.Duration) (workl
 	return sm, nil
 }
 
+func getLantecy(promAPI promv1.API, workloadName, workloadNamespace string, duration time.Duration, quantile float64) (time.Duration, error) {
+	latencyQuery := fmt.Sprintf(`histogram_quantile(%f, sum(rate(%s_bucket{%s=~"%s.*", %s=~"%s.*",reporter="destination"}[%s])) by (le))`,
+		quantile, reqDur, destWorkloadLabel, workloadName, destWorkloadNamespaceLabel, workloadNamespace, duration)
+
+	letency, err := vectorValue(promAPI, latencyQuery)
+	if err != nil {
+		return time.Duration(0), err
+	}
+
+	return convertLetencyToDuration(letency), nil
+}
+
 func vectorValue(promAPI promv1.API, query string) (float64, error) {
-	log.Debugf("executing query: %s", query)
 	val, _, err := promAPI.Query(context.Background(), query, time.Now())
 	if err != nil {
 		return 0, fmt.Errorf("query() failure for '%s': %v", query, err)
 	}
 
+	log.Debugf("executing query: %s  result:%s", query, val)
+
 	switch v := val.(type) {
 	case model.Vector:
 		if v.Len() < 1 {
 			log.Debugf("no values for query: %s", query)
 			return 0, nil
 		}
+
 		return float64(v[0].Value), nil
 	default:
 		return 0, errors.New("bad metric value type returned for query")
 	}
 }
 
+func convertLetencyToDuration(val float64) time.Duration {
```

Comment:
typo: should be `convertLatencyToDuration`

Reply:
fixed